configure.ac

```diff
@@ -1504,13 +1504,38 @@ case $host_os in
         ;;
 esac
 
+# MOLD section
+AC_ARG_ENABLE([mold],
+               AC_HELP_STRING([--enable-mold],
+                              [Use modern linker (mold) to link files]))
+
+USE_MOLD=no
+if test "x$enable_mold" = "xyes"; then
+   AC_CHECK_PROG([MOLDINSTALLED], [mold], [yes], [no])
+
+   if test "x$MOLDINSTALLED" = "xno"; then
+      AC_MSG_ERROR([`mold` is not installed, you need mold to link with it when using --enable-mold])
+   fi
+
+   AC_PATH_PROG([MOLDPATH], [ld], [no], [/usr/libexec/mold:/usr/local/libexec/mold])
```

Comment:
Ah that is the `mold` binary itself. How `-B` flag works is you provide it with a path where in substitutes for compiler's tools reside. Like if you check inside `/usr/local/libexec/mold` directory there is a single file called `ld` which is indeed a symlink to `/usr/local/bin/mold`. `gcc` sees that and uses our version of `ld` instead of the system one.

So, the path to pass with `-B` would be `/usr/local/libexec/mold` in case of source install and `/usr/libexec/mold` in case of repo install.

Regards